### PR TITLE
React-scripts to Vite

### PR DESCRIPTION
**Pull request by ap-justin**
### Notes when switching to this branch
- rename `.env` variables from `REACT_APP_` to `VITE_`

### Tooling changes

- remove all stuff related to `react-scripts` (`web-vitals`, `config-overrides (webpack)`)
- `REACT_APP_` variables renamed to `VITE_` but AWS counterparts remained `REACT_APP_`,
- `proces.env` converted to `import.meta.env`
- test runner changed from `jest` to `vitest`, but testing/assertion library is the same

### Source code changes

- rename `constants` folder into `constant` (without s) as `constants` conflicts with `node-modules/constants`
- remove `border-prim` utility as this messes up `postcss` for some reason (??) - expanded to raw `border-gray-l3  dark:border-bluegray`

### Others

- update Readme instructions
- change runtime in github actions
- update PR instructions

### CI speed comparison

| process      | react-scripts | vite |
| ------------ | --------------------- | -------- |
| lint         | 27s                   | 15s      |
| test         | 33s                   | 16s      |
| build        | 3m 2s                 | 1m 26s   |

## Instructions on making this work

- run `yarn` or `yarn install` to install npm dependencies
- run `yarn run test --watchAll` to verify all tests still pass
- (optional) run `yarn run build` to verify the build passes
- run `yarn start` to start the webapp
-

## UI changes for review

When major UI changes are introduced with a PR, please include links to URLS to compare or screenshots demonstrating the difference and notify on design changes

### CODE REVIEW

**Comment by ap-justin** [comment]
sers @SovereignAndrey ,@misicnenad, I've been updating this branch from time to time and has been ready for PR - I believe this is the right time to leave `react-scripts` as we currently have few draft branches that needs to reflect this upgrade.

**Comment by SovereignAndrey** [comment]
> sers @SovereignAndrey ,@misicnenad, I've been updating this branch from time to time and has been ready for PR - I believe this is the right time to leave `react-scripts` as we currently have few draft branches that needs to reflect this upgrade.

@ap-justin If you feel this is ready for a review, please take it out of "draft" and we'll start having a poke. :) 
One big consideration that comes to mind with this change is how bun would integrate with AWS Amplify's build & deployment process. I will try to read up on that today to see if possible and there are any major blockers on that front.

**Comment by ap-justin** [comment]
> One big consideration that comes to mind with this change is how bun would integrate with AWS Amplify's build & deployment process. I will try to read up on that today to see if possible and there are any major blockers on that front.

@SovereignAndrey , right amplify doesn't support bun yet - I have not considered this and got satisfied with this just working on github CI, I would try `node+yarn+vite` and see if there's much difference

**Review by SovereignAndrey — CHANGES_REQUESTED** [review]
@ap-justin I think the `amplify.yml`, `baseDirectory` location needs to be set as `dist` now with Vite vs `build` with React?

File: amplify.yml
Comment (SovereignAndrey) [inline amplify.yml:15]: `REACT_APP_APES_AUTH_SECRET_KEY` & `REACT_APP_ANGEL_AUTH_SECRET_KEY` should no longer be needed since AWS handles this via domain checks and CORS on the API gateway. cc: @jp-mariano

File: package.json
Comment (SovereignAndrey) [inline package.json:19]: As mentioned in Discord chat, let's keep this on v18 for now.

**Comment by ap-justin** [comment]
> One thing I noticed is that, the `amplify.yml`, `baseDirectory` location needs to be set as `dist` now with Vite vs `build` with React(I think?).

@SovereignAndrey, `outputDir` is still configured to be `build`
https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2385/files#diff-6a3b01ba97829c9566ef2d8dc466ffcffb4bdac08706d3d6319e42e0aa6890ddR21

File: package.json
Comment (ap-justin) [reply package.json]: fixed

File: amplify.yml
Comment (ap-justin) [reply amplify.yml:15]: i tried to remove these secrets but queries are failing e.g. marketplace cards doesn't load

Comment (SovereignAndrey) [reply amplify.yml:15]: From Localhost or something else? I don't think we only have allowlisted localhost on port 4200 domains setup in the API Gateway.

**Comment by SovereignAndrey** [comment]
> > One thing I noticed is that, the `amplify.yml`, `baseDirectory` location needs to be set as `dist` now with Vite vs `build` with React(I think?).

> @SovereignAndrey, `outputDir` is still configured to be `build` 

Ah! Okay, Got it, we are overriding the vite default there. Thank for pointing that out. :)

**Review by misicnenad — COMMENTED** [review]
Tentative approval, will leave it to Andrey to review and merge this.

@SovereignAndrey @ap-justin presumably we're moving to Vite because it's faster/stronger/better, but did I understand correctly that we're thinking of moving to Bun too? It's still a new tech that didn't stand the test of time.
It seems promising, don't get me wrong, but moving a production site to it seems risky.

**Comment by ap-justin** [comment]
> Tentative approval, will leave it to Andrey to review and merge this.
> 
> @SovereignAndrey @ap-justin presumably we're moving to Vite because it's faster/stronger/better,

@misicnenad, also so that we have more control over our deps, e.g we can now use typescript `v5` and upgrade tailwind without resorting to yarn resolutions

>  but did I understand correctly that we're thinking of moving to Bun too? It's a brand new tech that didn't stand the test of time.

we'll hold this upgrade prob until AWS adds bun runtime

File: amplify.yml
Comment (ap-justin) [reply amplify.yml]: removed from latest update in RC-v1.9

**Comment by ap-justin** [comment]
closing in favor of 
* #2617 

@SovereignAndrey @misicnenad

**Pull request closed without merging**
